lib/macho/macho_file.rb

```diff
@@ -233,10 +264,12 @@ def dylib_id=(new_id)
       raise ArgumentError.new("new ID must be a String") unless new_id.is_a?(String)
       return unless dylib?
 
-      dylib_cmd = command(:LC_ID_DYLIB).first
-      old_id = dylib_id
+      old_cmd = command(:LC_ID_DYLIB).first
+      new_cmd = LoadCommand.create(:LC_ID_DYLIB, new_id, old_cmd.timestamp,
+        old_cmd.current_version, old_cmd.compatibility_version)
```

Comment:
My intuition is that we should probably create one if it's missing, although I have an equally strong intuition that any dylib missing its ID is a huge warning sign and should probably be interpreted as an incorrectly formatted Mach-O.

I'll remove the ID from one of our test dylibs and see how OS X behaves with it and defer to that behavior :smile: